factory/interface.go

```diff
@@ -256,9 +256,10 @@ type NetworkComponentsHandler interface {
 	NetworkComponentsHolder
 }
 
-// TransactionSimulatorProcessor defines the actions which a transaction simulator processor has to implement
-type TransactionSimulatorProcessor interface {
-	ProcessTx(tx *transaction.Transaction) (*txSimData.SimulationResults, error)
+// TransactionCostSimulator defines the transaction cost simulator actions
+type TransactionCostSimulator interface {
```

Comment:
The main implementation for this is called TransactionCostEstimator, why not renaming the interface to `TransactionCostEstimator` ?

Reply:
refactored